File: mediapicker/src/main/java/iam/thevoid/mediapicker/builder/ImageIntentBuilder.java
package iam.thevoid.mediapicker.builder;

import android.content.Intent;

public class ImageIntentBuilder {

    private Mimetype mimetype = Mimetype.BOTH_IMAGE_AND_VIDEO;
    private boolean localOnly = false;

    public ImageIntentBuilder setLocalOnly(boolean localOnly) {
        this.localOnly = localOnly;
        return this;
    }

    public ImageIntentBuilder setMimetype(Mimetype mimetype) {
        this.mimetype = mimetype;
        return this;
    }

    public Intent build() {
        Intent intent = new Intent(Intent.ACTION_PICK);

        intent.setType(mimetype.getType());

        intent.setAction(Intent.ACTION_GET_CONTENT);

        intent.putExtra(Intent.EXTRA_LOCAL_ONLY, localOnly);

        return intent;
    }

    public enum Mimetype {
        IMAGE("image/*"),
        VIDEO("video/*"),
        BOTH_IMAGE_AND_VIDEO("image/*, video/*");

        Mimetype(String type) {
            this.type = type;
        }

        private String type;

        public String getType() {
            return type;
        }
    }
}

File: core/src/main/java/iam/thevoid/mediapicker/builder/ImageIntentBuilder.kt
package iam.thevoid.mediapicker.builder

import android.content.Intent

class ImageIntentBuilder {
    private var mimetype = MimeType.BOTH_IMAGE_AND_VIDEO
    private var localOnly = false

    fun setLocalOnly(localOnly: Boolean) = apply { this.localOnly = localOnly }

    fun setMimeType(mimeType: MimeType) = apply { this.mimetype = mimeType }

    fun build(): Intent = Intent(Intent.ACTION_PICK)
            .putExtra(Intent.EXTRA_LOCAL_ONLY, localOnly)
            .setType(mimetype.type)
            .setAction(Intent.ACTION_GET_CONTENT)

    enum class MimeType(val type: String) {
        IMAGE("image/*"),
        VIDEO("video/*"),
        BOTH_IMAGE_AND_VIDEO("image/*, video/*");
    }
}